File: 2022-2023/sourcecode/lab/lab13/OfflineCachingBasicsJava/app/src/main/java/pl/udu/uwr/pum/offlinecachingbasicsjava/data/adapter/UserComparator.java
package pl.udu.uwr.pum.offlinecachingbasicsjava.data.adapter;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.DiffUtil;

import pl.udu.uwr.pum.offlinecachingbasicsjava.data.User;

public class UserComparator extends DiffUtil.ItemCallback<User> {
    @Override
    public boolean areItemsTheSame(@NonNull User oldItem, @NonNull User newItem) {
        return newItem.getUid().equals(oldItem.getUid());
    }

    @Override
    public boolean areContentsTheSame(@NonNull User oldItem, @NonNull User newItem) {
        return newItem.getId() == oldItem.getId();
    }
}

File: 2022-2023/sourcecode/lab/lab13/OfflineCachingBasicsKotlin/app/src/main/java/pl/udu/uwr/pum/offlinecachingbasicskotlin/data/adapters/UserComparator.kt
package pl.udu.uwr.pum.offlinecachingbasicskotlin.data.adapters

import androidx.recyclerview.widget.DiffUtil
import pl.udu.uwr.pum.offlinecachingbasicskotlin.data.User

class UserComparator : DiffUtil.ItemCallback<User>() {
    override fun areItemsTheSame(oldItem: User, newItem: User): Boolean {
        return newItem.uid == oldItem.uid
    }

    override fun areContentsTheSame(oldItem: User, newItem: User): Boolean {
        return newItem == oldItem
    }
}